### PR TITLE
Updated depreciation

### PR DESCRIPTION
**Pull request by bruAristimunha**
As discussed in issue #412, this PR defers the deprecation of the scale function, and updates tutorials where the function is used.

### CODE REVIEW

File: examples/plot_bcic_iv_2a_moabb_cropped.py [hunk 1]
```diff
@@ -92,7 +93,7 @@
 
 preprocessors = [
     Preprocessor('pick_types', eeg=True, meg=False, stim=False),  # Keep EEG sensors
-    Preprocessor(scale, factor=1e6, apply_on_array=True),  # Convert from V to uV
+    Preprocessor(lambda data, factor: multiply(data, factor), factor=1e6),  # Convert from V to uV
```
Comment (robintibor) [inline examples/plot_bcic_iv_2a_moabb_cropped.py:96]:
```suggestion
    Preprocessor(partial(multiply, 1e6)),  # Convert from V to uV
```
should also work? Maybe it is nicer?
Also wondered if we could also forward *args to the function be applied and this way write something like
```
Preprocessor(multiply,2)
```
What do you think @bruAristimunha ?

Comment (bruAristimunha) [reply examples/plot_bcic_iv_2a_moabb_cropped.py:96]: I like the first option, using the partial, but this is definitely not a strong opinion.

Comment (bruAristimunha) [reply examples/plot_bcic_iv_2a_moabb_cropped.py:96]: whichever option you choose, I'm happy

Comment (bruAristimunha) [reply examples/plot_bcic_iv_2a_moabb_cropped.py:96]: What do you think @agramfort?

Comment (agramfort) [reply examples/plot_bcic_iv_2a_moabb_cropped.py:96]: No strong feeling. Partial is more pythonic but maybe harder for new comers. Now our custom API is not standard either. So as you prefer

Comment (bruAristimunha) [reply examples/plot_bcic_iv_2a_moabb_cropped.py:96]: Let's go for simple option, okay @robintibor?

**Review by agramfort — COMMENTED** [review]
guys I don't have much time these days. I let you approve and marge here.

**Pull request merged**